sound/soc/sof/intel/hda-loader.c

```diff
@@ -72,7 +72,7 @@ struct hdac_ext_stream *hda_cl_stream_prepare(struct snd_sof_dev *sdev, unsigned
 	hstream->format_val = format;
 	hstream->bufsize = size;
 
-	if (direction == SNDRV_PCM_STREAM_CAPTURE) {
+	if (direction == SNDRV_PCM_STREAM_CAPTURE && is_iccmax) {
```

Comment:
Reverse the check?

Reply:
The `else` will happily take iccmax && playback.
Should there be sanity check that iccmax flag is only allowed for capture stream?

Reply:
Good point indeed, thanks for the comment @ujfalusi .

Maybe only do `if (is_iccmax)` and not even check the direction?

@ranj063 what do you think?

Reply:
yes makes sense to check only is_iccmax.

Reply:
done